code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,470 @@
+/obj/item/xenoartifact/proc/create_beam(atom/target)
+	if((locate(src) in target?.contents) || !get_turf(target))
+		return
+	var/datum/beam/xenoa_beam/B = new(src.loc, target, time=1.5 SECONDS, beam_icon='icons/obj/xenoarchaeology/xenoartifact.dmi', beam_icon_state="xenoa_beam", btype=/obj/effect/ebeam/xenoa_ebeam)
+	B.set_color(material)
+	INVOKE_ASYNC(B, /datum/beam/xenoa_beam.proc/Start)
+
+///Default template used to interface with activator signals.
+/obj/item/xenoartifact/proc/default_activate(chr, mob/user, atom/target)
+	if(!COOLDOWN_FINISHED(src, xenoa_cooldown))
+		return FALSE
+	charge = chr
+	true_target |= process_target(target)
+	check_charge(user)
+	return TRUE
+
+///Add extra icon overlays. Ghetto GAGS.
+/obj/item/xenoartifact/proc/generate_icon(var/icn, var/icnst = "", colour)
+	icon_overlay = mutable_appearance(icn, icnst)
+	icon_overlay.layer = FLOAT_LAYER //Not doing this fucks the object icons when you're holding it
+	icon_overlay.appearance_flags = RESET_ALPHA// Not doing this fucks the alpha?
+	icon_overlay.alpha = alpha
+	if(colour)
+		icon_overlay.color = colour
+	add_overlay(icon_overlay)
+
+
+///Signaler traits. Sets listening freq
+/obj/item/xenoartifact/proc/set_frequency(new_frequency)
+	SSradio.remove_object(src, frequency)
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_XENOA)
+
+///Signaler traits. Sends signal
+/obj/item/xenoartifact/proc/send_signal(var/datum/signal/signal)
+	if(!radio_connection||!signal)
+		return
+	radio_connection.post_signal(src, signal)
+
+/obj/item/xenoartifact/receive_signal(datum/signal/signal)
+	if(!signal || signal.data["code"] != code)
+		return
+	SEND_SIGNAL(src, XENOA_SIGNAL, null, get_target_in_proximity(max_range), get_target_in_proximity(max_range)) //I don't think this sends a signal
+
+/obj/item/xenoartifact/on_block(mob/living/carbon/human/owner, atom/movable/hitby)
+	. = ..()
+	if(!(COOLDOWN_FINISHED(src, xenoa_cooldown)) || !get_trait(/datum/xenoartifact_trait/minor/blocking))
+		return
+	SEND_SIGNAL(src, COMSIG_PARENT_ATTACKBY, src, owner, hitby) //I don't think this sends a signal
+
+/obj/item/xenoartifact/process(delta_time)
+	switch(process_type)
+		if(PROCESS_TYPE_LIT) //Burning
+			true_target = list(get_target_in_proximity(min(max_range, 5)))
+			if(isliving(true_target[1]))
+				visible_message("<span class='danger' size='4'>The [name] flicks out.</span>")
+				default_activate(25, null, null)
+				process_type = null
+				return PROCESS_KILL
+		if(PROCESS_TYPE_TICK) //Clock-ing
+			playsound(get_turf(src), 'sound/effects/clock_tick.ogg', 50, TRUE) 
+			visible_message("<span class='danger' size='10'>The [name] ticks.</span>")
+			true_target = list(get_target_in_proximity(min(max_range, 5)))
+			default_activate(25, null, null)
+			if(prob(XENOA_TICK_CANCEL_PROB) && COOLDOWN_FINISHED(src, xenoa_cooldown))
+				process_type = null
+				return PROCESS_KILL
+		else    
```

Comment:
As arch said, there are spaces at the end of this line. (The original review could have been a little clearer)